cmd/livepeer/livepeer.go

```diff
@@ -142,6 +143,8 @@ func main() {
 	monitor := flag.Bool("monitor", false, "Set to true to send performance metrics")
 	version := flag.Bool("version", false, "Print out the version")
 	verbosity := flag.String("v", "", "Log verbosity.  {4|5|6}")
+	metadataAmqpUri := flag.String("metadataAmqpUri", "", "URI for AMQP host to send operation metadata")
```

Comment:
Looks like we took a slightly different approach here vs. the description in https://github.com/livepeer/go-livepeer/issues/1939. So, instead of having a generic metadata queue flag with the prefix of the URI indicating the protocol (i.e. AMQP) to be used, we're just using a AMQP specific metadata queue flag. This feels reasonable for now given that we're just supporting a single queue implementation, but it does seem like it would be nice to follow the same convention as the `-objectStore` flag where we can parse the prefix for a URI to determine which driver/implementation to use. Can consider for the future.

Reply:
So the main issue is that it is not possible to refer to a specific AMQP exchange or queue with a single URI. The URI can only go as far as to specify what's the server/cluster to connect and the `vhost` inside that server. Then since we needed a separate flag to specify a "queue" [1] already, I thought I should just make this amqp-specific as well. But I think we can do an intermediate solution here, have the URI be just `metadataUri` and we check if the protocol is `amqp(s)` before creating the AMQP producer. If not, we err, if yes, we create it with the other configurations provided in other flags. Same for any other protocol we give support to, which we can create specific flags for additional config if they need so, but this main URI can be kept protocol-agnostic 👌 

[1] instead of publishing events directly to a queue, I opted for an `exchange` instead which decouples `go-livepeer` from how the consumers want to consume the events. So the stream health node for example creates a stream, but some other service could create a regular queue if they want and process the events at-least-once, whatever.